utils/billgeneration.js

```diff
@@ -0,0 +1,112 @@
+/* eslint-disable import/imports-first */
+import moment from 'moment';
+import dotenv from 'dotenv-safe';
+
+import {
+  getSolarProjectById,
+  getOwnerById,
+  getRateScheduleById,
+  createSubscriberBill,
+  updateSubscriberBill,
+  getSubscriberBillsByStatementNumber
+} from '../airtable/request';
+import getEnphaseData from './enphase';
+import getLatestBill from './utilityApi';
+
+dotenv.config();
+
+const generateBillForSubscriber = async (subscriber, solarProject) => {
+  // Get data necessary to generate bill
+  console.log(`Generating Bill for ${subscriber.name}`);
+  const {
+    netPgeUsage,
+    ebceRebate,
+    startDate: startMoment,
+    endDate: endMoment
+  } = await getLatestBill(subscriber.meterId);
+
+  const startDate = startMoment.format('MM/DD/YYYY');
+  const endDate = endMoment.format('MM/DD/YYYY');
+
+  let prevBill;
+  if (subscriber.latestBillNumber !== 0) {
+    const bills = await getSubscriberBillsByStatementNumber(
+      subscriber.latestBillNumber
+    );
+    [prevBill] = bills.filter(b => b.subscriberId[0] === subscriber.id);
```

Comment:
im not familiar with this syntax - what does this do?

Reply:
It is called “Array Destructuring!” Honestly, I’ve never understood the purpose for it, but this was one of the few cases where I saw it’s usefulness.

`const [a, b] = [1, 2, 3]`
Is equivalent to
```
const array = [1, 2, 3];
const a = array[0;
const b = array[1];
```

So in this case it just gets the first element from the list